### DISCUSSION TITLE
[Host.Outbox] optimistic mode - hybrid strategy?

### DISCUSSION THREAD

**Discussion by dundich** [zarusz/SlimMessageBus#313] [Ideas]
Is it often the case that the bus is unavailable?
if the sequence of messages is not of particular interest. Maybe we should provide for writing to the outbox only when the bus is unavailable (exception) - and add an option with a hybrid strategy?

ps.  hybrid strategy -  just a “save -> don't save” checkbox in the outbox if the bus is unavailable

**Comment by zarusz** [comment]
Great idea, this falls under the curcuit breaker pattern for the outbox delivery.
Currently:
- assuming the DB is up the events will be save to the DB as usual
- but the background sending loop will fail these messages, when the target transport bus (ASB, Kafka, etc) is down (delivery attempts will increase)

A circuit breaker in outbox could detect the transport being down, pause the outbox send job, then probe if the transport bus is up, and if so resume delivery. 

If that is what would work for you - perhaps you could raise an issue for this feature?

There is something similar but for the consumers (outside of the outbox context): #282
